tests/fpga/fpga_vectorization_test.py

```diff
@@ -0,0 +1,751 @@
+                                           'target': dace.ScheduleType.FPGA_Device,
+                                           'strided_map': True
+                                       }) == 0
+
+    sdfg4: dace.SDFG = vecadd_1_non_appl_1_kernel.to_sdfg()
+    sdfg4.apply_transformations([FPGATransformSDFG, InlineSDFG])
+
+    assert sdfg4.apply_transformations(Vectorization,
+                                       options={
+                                           'vector_len': 2,
+                                           'target': dace.ScheduleType.FPGA_Device,
+                                           'strided_map': False
+                                       }) == 0
+
+    assert sdfg4.apply_transformations(Vectorization,
+                                       options={
+                                           'vector_len': 2,
+                                           'target': dace.ScheduleType.FPGA_Device,
+                                           'strided_map': True
+                                       }) == 0
+
+    sdfg5: dace.SDFG = maporder_vec_kernel.to_sdfg()
+
+    sdfg5.apply_transformations([FPGATransformSDFG, InlineSDFG])
+
+    assert sdfg5.apply_transformations(Vectorization,
+                                       options={
+                                           'vector_len': 2,
+                                           'target': dace.ScheduleType.FPGA_Device,
+                                           'strided_map': False
+                                       }) == 0
+
+    assert sdfg5.apply_transformations(Vectorization,
+                                       options={
+                                           'vector_len': 2,
+                                           'target': dace.ScheduleType.FPGA_Device,
+                                           'strided_map': True
+                                       }) == 0
+
+    sdfg6: dace.SDFG = matmul_vec_kernel.to_sdfg()
+
+    sdfg6.apply_transformations([FPGATransformSDFG, InlineSDFG])
+
+    assert sdfg6.apply_transformations(Vectorization,
+                                       options={
+                                           'vector_len': 2,
+                                           'target': dace.ScheduleType.FPGA_Device,
+                                           'strided_map': False
+                                       }) == 0
+
+    assert sdfg6.apply_transformations(Vectorization,
+                                       options={
+                                           'vector_len': 2,
+                                           'target': dace.ScheduleType.FPGA_Device,
+                                           'strided_map': True
+                                       }) == 0
+
+
+@fpga_test()
+def test_vec_two_maps_strided():
+    return vec_two_maps(True)
+
+
+@fpga_test()
+def test_vec_two_maps_non_strided():
+    return vec_two_maps(False)
+
+
+@xilinx_test()
```

Comment:
For the ones that only work for Xilinx, can you make a note about why it fails for Intel?

Reply:
It is always the 'auto' keyword which causes the error: ` error: OpenCL C version 1.2 does not support the 'auto' storage class specifier`